src/main/java/filter/expression/Qualifier.java

```diff
@@ -93,6 +93,40 @@ public Qualifier(String name, List<SortKey> keys) {
         this.sortKeys = new ArrayList<>(keys);
     }
 
+    public static FilterExpression replaceMilestoneAliases(IModel model, FilterExpression expr) {
+        List<String> repoIds = getContentOfMetaQualifier(expr, REPO).stream()
+                .map(repoId -> repoId.toLowerCase())
+                .collect(Collectors.toList());
+
+        if (repoIds.isEmpty()) {
+            repoIds.add(model.getDefaultRepo().toLowerCase());
+        }
+
+        List<TurboMilestone> allMilestones = model.getMilestones().stream()
+                .filter(ms -> repoIds.contains(ms.getRepoId().toLowerCase()))
+                .sorted(Qualifier::milestoneDueDateComparator)
+                .collect(Collectors.toList());
+
+        Optional<Integer> currentMilestoneIndex = getCurrentMilestoneIndex(allMilestones);
```

Comment:
Maybe we could check this here and return `expr` if it's absent? We don't need to do the rest if there is some index, and the rest can just deal with an int.